bpf/bpf_host.c

```diff
@@ -378,7 +379,14 @@ handle_ipv6_cont(struct __ctx_buff *ctx, __u32 secctx, const bool from_host,
 	info = lookup_ip6_remote_endpoint(dst, 0);
 
 #ifdef TUNNEL_MODE
-	if (info && info->flag_skip_tunnel)
+	__u32 src_subnet_id = lookup_ip6_subnet((union v6addr *)&ip6->saddr);
+	__u32 dst_subnet_id = lookup_ip6_subnet((union v6addr *)&ip6->daddr);
+	bool same_subnet = (src_subnet_id == dst_subnet_id) && (src_subnet_id != 0);
+
+	cilium_dbg3(ctx, DBG_TUNNEL_TRACE, src_subnet_id, dst_subnet_id, src_subnet_id);
+	cilium_dbg3(ctx, DBG_TUNNEL_TRACE, src_subnet_id, dst_subnet_id, dst_subnet_id);
+	cilium_dbg3(ctx, DBG_SUBNET_CHECK, src_subnet_id, dst_subnet_id, same_subnet);
```

Comment:
I'm not a big fan of putting debug hook on this kind of hot path even if it can be opted out. Do you think we really need this here? Same for IPv4 path.

Reply:
This is for debugging purposes only, will be removed before merging. Keeping this open to make sure I do remove it - https://github.com/cilium/cilium/pull/41868#discussion_r2373845980 :-)

Reply:
I have removed it now.